dora/core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -710,30 +712,56 @@ public boolean rename(Path src, Path dst) throws IOException {
     } catch (InvalidArgumentRuntimeException e) {
       throw new IllegalArgumentException(e);
     } catch (AlluxioRuntimeException e) {
-      throw toHdfsIOException(e);
+      LOG.warn("rename failed: {}", toHdfsIOException(e));
+      return renameInternal(src, dst, srcPath, dstPath);
     } catch (AlluxioException e) {
+      return renameInternal(src, dst, srcPath, dstPath);
```

Comment:
i had to write try-catch inside this catch but i figure it's ok
```
    } catch (AlluxioRuntimeException | AlluxioException e) {
      Exception exToLog = e;
      if (e instanceof AlluxioRuntimeException) {
        exToLog = toHdfsIOException((AlluxioRuntimeException) e);
      }
      try {
        boolean res = tryMoveIntoDirectory(srcPath, dstPath);
        if (!res) {
          LOG.warn("Failed to rename file and dst {} is not a directory", dst, exToLog);
        }
        return res;
      } catch (IOException | AlluxioException | AlluxioRuntimeException e2) {
        LOG.error("Failed to rename file {} and failed to move into dst directory {}",
            src, dst, exToLog);
        return false;
      }
```

Reply:
can we package the try-catch into the `tryMoveIntoDirectory`? then we can reduce the nested try-catch.

Reply:
agree

Reply:
i just wanted to have the ex from the 1st try `AlluxioRuntimeException | AlluxioException e` and the ex from the `tryMoveIntoDirectory` together. So I can decide which one to log, and maybe log both?